### PR TITLE
Rework HT16K33 LED/keyscan support

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
This PR reworks the HT16K33 LED/keyscan support
- It converts the keyscan sub-driver from the GPIO API to the kscan API (which was not present when this driver was introduced).
- It fixes an issue with failed I2C communication during IRQ handling, causing no further IRQs to be delivered.
- Converts the driver from using `device_get_binding()` to the newer DT macros.

### CODE REVIEW

**Comment by henrikbrixandersen** [comment]
Once this PR is merged, I will follow up with a generic HT16K33 shield definition and rework `tests/drivers/build_all` with proper handling of the HT16K33.

**Review by mnkp — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]
Seems reasonable, though I do not know the kscan api

File: drivers/led/ht16k33.c [hunk 1]
```diff
@@ -123,7 +118,7 @@ static int ht16k33_led_blink(const struct device *dev, uint32_t led,
 		cmd |= HT16K33_OPT_BLINK_2HZ;
 	}
 
-	if (i2c_write(data->i2c, &cmd, 1, config->i2c_addr)) {
+	if (i2c_write(config->i2c_dev, &cmd, 1, config->i2c_addr)) {
```
Comment (VenkatKotakonda) [inline drivers/led/ht16k33.c:121]: instead of hardcoded size 1, please use sizeof()

Comment (henrikbrixandersen) [reply drivers/led/ht16k33.c]: Fixed.

**Review by VenkatKotakonda — APPROVED** [review]

**Pull request merged**